src/main/java/com/example/SOPSbackend/service/AdminService.java

```diff
@@ -1,29 +1,110 @@
 package com.example.SOPSbackend.service;
 
+import com.example.SOPSbackend.dto.EditDoctorDto;
+import com.example.SOPSbackend.dto.EditPatientDto;
+import com.example.SOPSbackend.model.AddressEntity;
+import com.example.SOPSbackend.model.BasicUserEntity;
 import com.example.SOPSbackend.model.DoctorEntity;
+import com.example.SOPSbackend.model.PatientEntity;
 import com.example.SOPSbackend.repository.DoctorRepository;
+import com.example.SOPSbackend.repository.PatientRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.annotation.Secured;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
 
+/**
+ *  <p>I know this is questionable,
+ *  but it's been decided that
+ *  we are using the AdminService class
+ *  for "everything admin"
+ *  (everything related to the admins' endpoints).</p>
+ *
+ *  <p> We could use DoctorService
+ *  to modify doctors tables,
+ *  but the currently implemented option
+ *  seems more logical and clean.</p>
+ *
+ *  <p>Any and all thoughts about this
+ *  are appreciated at
+ *  filip.spychala.stud [at symbol] pw.edu.pl.</p>
+ */
 @Service
+@Transactional
+@Secured({"ROLE_ADMIN"}) //TODO: check if that prevents other users from using these requests
 public class AdminService {
+    private static final int ITEMS_PER_PAGE = 10;
```

Comment:
Chyba bym uwspólnił, żeby każdy controller/service miał to samo
Ale no, nie wiem czy

Reply:
? @V1OL3TF0X